frontend/packages/dev-console/src/utils/resource-label-utils.ts

```diff
@@ -92,11 +94,16 @@ export const mergeData = (originalResource: K8sResourceKind, newResource: K8sRes
     mergedData.spec.template.spec.containers = newResource.spec.template.spec.containers;
   }
   if (mergedData?.spec?.hasOwnProperty('strategy')) {
-    mergedData.spec.strategy = newResource.spec?.strategy ?? originalResource?.spec?.strategy;
+    mergedData.spec.strategy = newResource.spec?.strategy ?? originalResource.spec?.strategy;
   }
   if (mergedData.spec?.triggers) {
     mergedData.spec.triggers = newResource.spec.triggers;
   }
+  if (mergedData.spec?.template?.spec?.hasOwnProperty('volumes')) {
```

Comment:
Still a little perplexed by why we need this... This feels like we won't really catch everything that one could add to Deployments.

Essentially this solution feels like it will close the specific ask we had in the bug but not address the underlying issue of "parameters being lost if we are not explicitly calling them out". 🤔

Reply:
@andrewballantyne the issue here is that we dont have a way for adding/editing volumes/volumeMounts from the create/edit app forms but volumes/volumeMounts can be added/updated directly in the yaml itself so currently the solution that makes sense to me is to preserve whatever volumes/volumeMounts we have in the originalResource.

Reply:
I understand the reason for the change -- what I have confusion around is the way the fix is implemented. I know this method has other statements like this, but it seems we are putting a bandaid on a stab wound. We are not allowing properties to flow through (like a spread or merge of objects), so we are stuck to manually green lighting each property as someone raises it to us. 

Not sure we need to address this issue here, it's a general issue with the edit flow in general today for all the resources we maintain. But it's definitely concerning.

Reply:
Do you have a suggestion on how do we handle this here?

Reply:
Do you have a suggestion on how do we handle this here?

Reply:
Not without more tests 🙂 

Essentially you are already merging the two objects, I think if we deep merge it ... it MIGHT solve our issue. But I am a bit concerned about breaking everything else haha. I don't think this is the best time to do it. We can fix the direct issue now and then look to tighten up things in the future.

Reply:
@divyanshiGupta It occurred again, I think we need to go through with this now.
RE: https://bugzilla.redhat.com/show_bug.cgi?id=1929577

Reply:
Yes, it's because how these statements got added without considering what happens to the old values. This particular issue is occurring because the old containers are directly replaced with what we generate not taking in account specific container attribute values like envs and that these values can be added from elsewhere so we cannot override everything.
https://github.com/openshift/console/pull/4517/files#diff-7fd530a4a1b4113f74f1e7e64beadbdb448845c1fad06d6a4d4824425583c2d6R53-R55

I think we should create a tech debt for this and then I can work on refactoring this function and add some unit tests.

Reply:
https://issues.redhat.com/browse/ODC-5539

Reply:
We could also fix it part of the BZ I logged... this is the 2nd customer bug we have on this function; it needs to be fixed sooner rather than later.